src/basic/locale-util.c

```diff
@@ -109,7 +109,8 @@ static int add_locales_from_archive(Set *locales) {
         if (!S_ISREG(st.st_mode))
                 return -EBADMSG;
 
-        if (st.st_size < (off_t) sizeof(struct locarhead))
+        if (st.st_size < (off_t) sizeof(struct locarhead) ||
+            (uintmax_t) st.st_size > SIZE_MAX)
```

Comment:
nit: this can all be on one line. no need to break lines too eagerly

Reply:
done